csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -264,14 +265,20 @@ private static DisposableNamedOnnxValue DisposableNamedOnnxValueFromNativeTensor
 
         protected virtual void Dispose(bool disposing)
         {
+            if(_disposed)
+            {
+                return;
+            }
+
+            // dispose managed state (managed objects).
             if (disposing)
             {
-                // dispose managed state (managed objects).
                 if (_nativeMemoryManager != null)
                 {
                     _nativeMemoryManager.Dispose();
                     _nativeMemoryManager = null;
                 }
+                _disposed = true;
             }
```

Comment:
nit: this should be outside of the 'if (disposing)' according to [https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1063?view=vs-2019](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1063?view=vs-2019)

Reply:
You mean it should be INSIDE?

---
In reply to: [477592821](https://github.com/microsoft/onnxruntime/pull/4915#discussion_r477592821) [](ancestors = 477592821)